ToolBox/PsychSignalDataConverter/PsychSignalDataConverter.cs

```diff
@@ -0,0 +1,270 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using QuantConnect.Logging;
+using QuantConnect.Util;
+
+namespace QuantConnect.ToolBox.PsychSignalDataConverter
+{
+    public class PsychSignalDataConverter
+    {
+        private readonly Dictionary<string, TickerData> _fileHandles;
+        private readonly HashSet<string> _knownTickers;
+        private readonly DirectoryInfo _rawSourceDirectory;
+        private readonly DirectoryInfo _destinationDirectory;
+
+        /// <summary>
+        /// Converts psychsignal raw data into a format usable by Lean
+        /// </summary>
+        /// <param name="sourceDirectory">Directory to source our raw data from</param>
+        /// <param name="destinationDirectory">Directory to write formatted data to</param>
+        /// <param name="knownTickerFolder">Directory where we source the ticker list</param>
+        public PsychSignalDataConverter(string sourceDirectory, string destinationDirectory, string knownTickerFolder)
+        {
+            _rawSourceDirectory = new DirectoryInfo(sourceDirectory);
+            _destinationDirectory = new DirectoryInfo(destinationDirectory);
+
+            _fileHandles = new Dictionary<string, TickerData>();
+            _knownTickers = Directory.GetFiles(knownTickerFolder, "*.zip").Select(Path.GetFileNameWithoutExtension).ToHashSet();
+
+            _destinationDirectory.Create();
```

Comment:
I'm torn on things like this in constructors. On one hand, it's nice to encapsulate things like this inside the class, but on the other hand, it's  typically best to avoid potentially exception-raise actions inside constructors. No need to change, just thinking (writing) out loud :)